tests/chainer_tests/functions_tests/math_tests/test_zeta.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+
+import numpy
+
+import chainer
+import chainer.functions as F
+from chainer import testing
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32, numpy.float64]
+}))
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [
+        {},
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'cuda_device': [0, 1],
+    })
+    # ChainerX tests
+    + testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0', 'cuda:1']
+    })
+)
+@testing.with_requires('scipy')
+class TestZeta(testing.FunctionTestCase):
+
+    def setUp(self):
+        self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-3}
+        self.check_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
+        self.check_double_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
+
+        self._x = numpy.array(numpy.random.uniform(5, 10))
+
+    def generate_inputs(self):
+        q = numpy.array(numpy.random.uniform(1, 10))
```

Comment:
Do you have to wrap `q` and `self._x` above in `numpy.array`? It seems redundant.

Reply:
Ah, how about parameterizing over shapes similar to `test_polygamma.py`. An also, please don't forget to cast to `self.dtype` (again see `test_polygamma.py`).

Reply:
Removed wrapping and added shape parametrization